distributed/tests/test_cancelled_state.py

```diff
@@ -602,8 +607,10 @@ async def release_all_futures():
                 (f3.key, "ready", "executing", "executing", {}),
                 (f3.key, "executing", "released", "cancelled", {}),
                 (f3.key, "cancelled", "fetch", "resumed", {}),
-                (f3.key, "resumed", "waiting", "executing", {}),
+                (f3.key, "resumed", "released", "cancelled", {}),
+                (f3.key, "cancelled", "waiting", "executing", {}),
                 (f3.key, "executing", "error", "error", {}),
+                # FIXME: (distributed#7489)
```

Comment:
Instead of accepting the erred task, the scheduler should reject the result and reschedule the computation (#7489)